hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/compress/CompressDecompressTester.java

```diff
@@ -432,7 +412,11 @@ public void assertCompression(String name, Compressor compressor,
               joiner.join(name, "byte arrays not equals error !!!"),
               originalRawData, decompressOut.toByteArray());
         } catch (Exception ex) {
-          fail(joiner.join(name, ex.getMessage()));
+          if (ex.getMessage() != null) {
+            fail(joiner.join(name, ex.getMessage()));
+          } else {
+            fail(joiner.join(name, ExceptionUtils.getStackTrace(ex)));
```

Comment:
Why is this change needed?

Reply:
When I first took over this change, the test failed with NPE without any details. It is because the exception thrown returns null from `getMessage()`. `joiner.join(name, null)` causes the NPE, so I changed it to print stack trace once `getMessage()` returns null. It's better for debugging.

Reply:
NPE is why toString() is what new code should do.
Why don't we just `throw new AssertionError(name +ex, ex)`. That way, the stack trace doesn't get lost, which is something we never want to have happen,